src/overlays/actors/ovl_En_Ssh/z_en_ssh.c

```diff
@@ -708,11 +708,11 @@ void EnSsh_Talk(EnSsh* this, PlayState* play) {
 void func_809756D0(EnSsh* this, PlayState* play) {
     u16 phi_a1;
 
-    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_34_08)) {
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_MASK_OF_TRUTH)) {
         phi_a1 = 0x914;
     } else {
         phi_a1 = 0x910;
-        SET_WEEKEVENTREG(WEEKEVENTREG_34_08);
+        SET_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_MASK_OF_TRUTH);
```

Comment:
`WEEKEVENTREG_RECEIVED_MASK_OF_TRUTH` can be set in two places. One right where the GI_MASK_TRUTH is given, and here. Would you know what's the difference?

Reply:
I see one set and one clear, and no the flags in this actor are confusing enough I didnt name them all because its not that clear to me, I have to assume there is some case where the player starts dialogue but leaves the room or something.

Reply:
The other set is here, in a different file:
https://github.com/zeldaret/mm/blob/08f6b1c24bbfb32103fa3e8a40e37704d7946906/src/overlays/actors/ovl_En_Sth/z_en_sth.c#L520-L521
And yeah, no worries about the flags. Seems like these files are all intermingled.

Reply:
Oh thats even weirder, I was looking at Sth first not thinking about Ssh, why would Ssh _ever_ set the flag?? Thats extra weird.

Reply:
The clear of `WEEKEVENTREG_RECEIVED_MASK_OF_TRUTH` actually seems weird to me. Why would that flag be cleared? Wouldn't that mean you'd get mask of truth multiple times if you redo the swamp house? What's preventing that?

Reply:
Seems like Ssh uses the flag to detect if the player as already talked to them once when entering the house, Sth might be re-using it differently, under the assumption that only one of them can exist at any point at how Ssh used it is no longer relevant, let me test something